src/Microsoft.NetCore.Analyzers/Core/Security/SourceTriggeredTaintedDataAnalyzerBase.cs

```diff
@@ -77,13 +83,81 @@ public override void Initialize(AnalysisContext context)
                                 operationAnalysisContext =>
                                 {
                                     IInvocationOperation invocationOperation = (IInvocationOperation)operationAnalysisContext.Operation;
-                                    if (sourceInfoSymbolMap.IsSourceMethod(invocationOperation.TargetMethod))
+                                    bool needValueContentAnalysis = false;
+                                    if (sourceInfoSymbolMap.IsSourceMethod(invocationOperation.TargetMethod, out var evaluateWithPointsToAnalysis, out var evaluateWithValueContentAnslysis))
                                     {
-                                        rootOperationsNeedingAnalysis.Add(operationAnalysisContext.Operation.GetRoot());
+                                        IOperation rootOperation = operationAnalysisContext.Operation.GetRoot();
+                                        PointsToAnalysisResult pointsToAnalysisResult;
+                                        ValueContentAnalysisResult valueContentAnalysisResultOpt;
+                                        if (evaluateWithPointsToAnalysis.Count != 0)
+                                        {
+                                            pointsToAnalysisResult = PointsToAnalysis.TryGetOrComputeResult(
+                                                rootOperation.GetEnclosingControlFlowGraph(),
+                                                owningSymbol,
+                                                WellKnownTypeProvider.GetOrCreate(operationAnalysisContext.Compilation),
+                                                InterproceduralAnalysisConfiguration.Create(
+                                                    operationAnalysisContext.Options,
+                                                    SupportedDiagnostics,
+                                                    defaultInterproceduralAnalysisKind: InterproceduralAnalysisKind.ContextSensitive,
+                                                    cancellationToken: operationAnalysisContext.CancellationToken),
+                                                interproceduralAnalysisPredicateOpt: null);
+                                            if (pointsToAnalysisResult == null
+                                                || !evaluateWithPointsToAnalysis.Any(s => s(
+                                                    invocationOperation.Arguments.Select(o => pointsToAnalysisResult[o.Kind, o.Syntax]))))
+                                            {
+                                                return;
+                                            }
+                                        }
+                                        else if (evaluateWithValueContentAnslysis.Count != 0)
+                                        {
+                                            valueContentAnalysisResultOpt = ValueContentAnalysis.TryGetOrComputeResult(
```

Comment:
>ValueContentAnalysis.TryGetOrComputeResult [](start = 76, length = 42)

ValueContentAnalysis doesn't cache its results.  Can you change that in this PR?

@mavasani, I assume that's fine since security analyzers are making use of it. 😀

Reply:
Yes